packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -309,11 +312,14 @@ export class DefaultWeb3Handler implements Web3Handler, FullnodeHandler {
     address: Address,
     defaultBlock: string
   ): Promise<string> {
-    if (defaultBlock !== 'latest') {
+    const curentBlockNumber = await this.provider.getBlockNumber()
+    if (
+      !['latest', numberToHexString(curentBlockNumber)].includes(defaultBlock)
+    ) {
       log.debug(
         `No support for historical code lookups! Anything returned from this may be very wrong.`
       )
-      //throw new Error('No support for historical code lookups!')
+      throw new Error('No support for historical code lookups!')
```

Comment:
Can you add `currentBlockNumber` and `defaultBlock` to the log and error messages?